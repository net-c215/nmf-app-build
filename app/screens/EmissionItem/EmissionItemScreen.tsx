import React from "react";
import { useSelector } from "react-redux";
import { ScrollView } from "react-native";
import { pathOr } from "ramda";
import moment from "moment";
import { t } from "../../utils";
import styles from "./EmissionItemScreen.styles";
import { Text, Tag } from "../../components";
import { selectors } from "./ducks";
import navigationOptions from "./EmissionItemScreen.navigationOptions";

interface Props {
  navigation: {
    state: {
      params: {
        id: string;
      };
    };
    goBack: () => void;
  };
}

const EmissionItemScreen = ({ navigation }: Props) => {
  const emissionId = pathOr(false, ["state", "params", "id"], navigation);

  if (!emissionId) {
    navigation.goBack();
  }

  const emission = useSelector(state =>
    selectors.getEmissionItemById(state, emissionId)
  );
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const onPress = () => {};
  const date = moment(emission.creationDate, "YYYY-MM-DDTHH:mm:ss.sssZ").format(
    "dddd, MMMM Do YYYY"
  );
  console.log("TCL: EmissionItemScreen -> emission", emission);

  return (
    <ScrollView style={styles.container}>
      <Text.H3>{t("EMISSION_ITEM_TYPE")}</Text.H3>
      <ScrollView horizontal style={styles.item}>
        <Tag selected onPress={onPress} title={emission.emissionModelType} />
      </ScrollView>
      <Text.H3>{t("EMISSION_ITEM_QUANTITY")}</Text.H3>
      <Text.Primary darkGray style={styles.item}>
        {emission.value}
      </Text.Primary>
      <Text.H3>{t("EMISSION_ITEM_DATE")}</Text.H3>
      <Text.Primary darkGray style={styles.item}>
        {date}
      </Text.Primary>
    </ScrollView>
  );
};

EmissionItemScreen.navigationOptions = navigationOptions;

export default EmissionItemScreen;
