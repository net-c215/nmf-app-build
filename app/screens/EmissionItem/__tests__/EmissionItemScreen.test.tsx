import React from "react";
import renderer from "react-test-renderer";
import EmissionItemScreen from "../EmissionItemScreen";

it("EmissionsScreen renders correctly", () => {
  const props = {
    navigation: {
      goBack: jest.fn(),
      state: {
        params: {
          id: "ejew034-dsd"
        }
      }
    }
  };
  const tree = renderer.create(<EmissionItemScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
