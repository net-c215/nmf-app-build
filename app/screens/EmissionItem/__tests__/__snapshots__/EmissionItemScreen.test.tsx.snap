// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmissionsScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
      "paddingTop": 22,
    }
  }
>
  <View>
    <Text.H3>
      Type
    </Text.H3>
    <RCTScrollView
      horizontal={true}
      style={
        Object {
          "paddingBottom": 24,
          "paddingTop": 6,
        }
      }
    >
      <View>
        <Tag
          onPress={[Function]}
          selected={true}
        />
      </View>
    </RCTScrollView>
    <Text.H3>
      Quantity
    </Text.H3>
    <Text.Primary
      darkGray={true}
      style={
        Object {
          "paddingBottom": 24,
          "paddingTop": 6,
        }
      }
    />
    <Text.H3>
      Date
    </Text.H3>
    <Text.Primary
      darkGray={true}
      style={
        Object {
          "paddingBottom": 24,
          "paddingTop": 6,
        }
      }
    >
      Invalid date
    </Text.Primary>
  </View>
</RCTScrollView>
`;
